PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/AdUnit.java

```diff
@@ -449,6 +477,8 @@ public void onFetchCompleted(BidResponse response) {
                 HashMap<String, String> keywords = response.getTargeting();
                 Util.apply(keywords, adObject);
                 originalListener.onComplete(ResultCode.SUCCESS);
+
+                registerVisibilityTrackerIfNeeded(bidResponse);
```

Comment:
This approach doesn't look reliable. Switching the tracker on/off for each result of the bid fetching may lead to gaps in tracking.

Let's implement it in the way introduced on iOS.

- start viewability tracking on calling of `activatePrebidImpressionTracker` 
- stop tracking on destroying the AdUnit object 
- once the tracked view meets the impression tracking criteria - check the content, and if it contains PUC - track the respective impression.